zcash_client_sqlite/src/testing.rs

```diff
@@ -211,6 +216,12 @@ impl<Cache> TestBuilder<Cache> {
         self
     }
 
+    pub(crate) fn with_account_having_index(mut self, index: zip32::AccountId) -> Self {
```

Comment:
This method name is misleading: it only works if either `with_account_having_current_birthday` or `with_account_from_sapling_activation` was called, and its effect is to _modify_ the account index of the account that those methods would create. Rename the method to something clearer and document it.